### PR TITLE
 Pull in shakedown as an editable dependency. 

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
We have a lot of Shakedown changes pending in `tests/system/common.py`.
So far it takes a few steps to move the changes to Shakedown and then
pull them back again. This patch eases the process by importing
Shakedown as a git-subrepo. It allows making changes to Shakedown in the
Marathon repository *and* push changes upstream to create pull requests
for Shakedown. This way Shakedown in Marathon will be slightly ahead of
the Shakedown master *but* also ensure that the Shakedown master will
stay up-to-date.

### CODE REVIEW

**Comment by jeschkies** [comment]
This is a proposal of mine. Please take a quick look at [git-subrepo](https://github.com/ingydotnet/git-subrepo#readme). It's less scary than it seems.

If you are using Shakedown in Marathon you have to do nothing except call the tests with Pipenv as the CI already does:

```
cd tests/system && make test
```

or 

```
cd tests/system
pipenv shell
shakedown test_marathon_root.py::<test name>
```

or

```
cd tests/system
pipenv run shakedown test_marathon_root.py::<test name>
```

If you want to make edits to Shakedown simply edit the files in `tests/shakedown` and commit them to the Marathon repo.

Once these commits landed or event before you can push the changes to a Shakedown branch with
```
git subrepo push tests/shakedown -b my-feature
```

I guess that only applies for Shakedown contributors.

If we want changes from Shakedown we can pull them with

```
git subrepo pull tests/shakedown -b master
```

`git-subrepo` will squash upstream patches in one update commit to keep Marathons commit history clean.

This way we get both benefits. A local Shakedown for fast iteration and upstream patches.

File: tests/README.md [hunk 1]
```diff
@@ -71,3 +71,26 @@ shakedown test_marathon_root.py::<test name>
 
 For more details checkout the [shakedown site](https://github.com/dcos/shakedown).
 The tests are written under the project [test/system](system/README.md).
+
+### Shakedown Patching
```
Comment (jeschkies) [inline tests/README.md]: This is the only edit I made.

File: tests/shakedown/.gitrepo [hunk 2]
```diff
@@ -0,0 +1,11 @@
+; DO NOT EDIT (unless you know what you are doing)
```
Comment (kensipe) [inline tests/shakedown/.gitrepo]: shouldn't this always be true :)

File: tests/system/marathon_common_tests.py [hunk 3]
```diff
@@ -826,14 +826,14 @@ def test_app_update_rollback():
 
     client = marathon.create_client()
     client.add_app(app_def)
-    shakedown.deployment_wait()
+    shakedown.deployment_wait(app_id=app_id)
```
Comment (kensipe) [inline tests/system/marathon_common_tests.py]: I would not conflate these changes with the subrepo change

Comment (jeschkies) [reply tests/system/marathon_common_tests.py]: Yep. You are right. I won't merge this until https://github.com/mesosphere/marathon/pull/6302 landed.

**Pull request merged**